### PR TITLE
feat(setup-elixir): support version-file

### PR DESCRIPTION
**Pull request by danewalters**

### CODE REVIEW

File: setup-elixir/action.yml [hunk 1]
```diff
@@ -1,43 +1,60 @@
...
   submodules:
     description: Whether to checkout submodules
     default: false
+  use-tool-versions:
+    description: Whether to use .tool-versions file for version management
+    default: false
   elixir-version:
     description: Version range or exact version of Elixir
-    required: true
+    required: false
```
Comment (danewalters) [inline setup-elixir/action.yml:16]: 这里需不需要设置一个默认版本哇 @fahchen

Comment (danewalters) [reply setup-elixir/action.yml:16]: required 为 false，string 类型不设置默认是一个空字符串

File: setup-elixir/action.yml [hunk 2]
```diff
@@ -1,43 +1,60 @@
...
 description: Setup a specific version of Erlang/OTP and Elixir
+
 inputs:
   fetch-depth:
     description: Number of commits to fetch
     default: 1
   submodules:
     description: Whether to checkout submodules
     default: false
+  use-tool-versions:
```
Comment (fahchen) [inline setup-elixir/action.yml]: 保持一致
```suggestion
  tool-versions:
```

File: setup-elixir/action.yml (hunk 1 above)
Comment (fahchen) [reply setup-elixir/action.yml:16]: 不用给默认值

File: setup-elixir/action.yml [hunk 3]
```diff
@@ -1,43 +1,60 @@
...
       with:
         fetch-depth: ${{ inputs.fetch-depth }}
         submodules: ${{ inputs.submodules }}
-    - uses: erlef/setup-beam@v1
+
+    - name: Setup Erlang/OTP and Elixir using .tool-versions
+      if: ${{ inputs.use-tool-versions == 'true' }}
+      uses: erlef/setup-beam@v1
+      with:
+        version-file: '.tool-versions'
```
Comment (fahchen) [inline setup-elixir/action.yml]: https://github.com/erlef/setup-beam?tab=readme-ov-file#version-file
测试下，把成功的 run 贴一下
```suggestion
        version-file: '.tool-versions'
        version-type: strict
```

File: setup-elixir/action.yml [hunk 4]
```diff
@@ -1,43 +1,60 @@
...
+
+    - name: Setup Erlang/OTP and Elixir from input versions
+      if: ${{ inputs.use-tool-versions == 'false' }}
+      uses: erlef/setup-beam@v1
       with:
         otp-version: ${{ inputs.otp-version }}
         elixir-version: ${{ inputs.elixir-version }}
```
Comment (fahchen) [inline setup-elixir/action.yml]: 你试试看这两步能不能合并到一起？
```
      - uses: erlef/setup-beam@v1
        with:
          version-file: ${{ inputs.version-file }}
          version-type: strict
          otp-version: ${{ inputs.otp-version }}
          elixir-version: ${{ inputs.elixir-version }}
```

**Review by fahchen — CHANGES_REQUESTED** [review]

File: setup-elixir/action.yml (hunk 3 above)
Comment (danewalters) [reply setup-elixir/action.yml]: https://github.com/Byzanteam/jet-storage-plugin/actions/runs/10980131585

File: setup-elixir/action.yml [hunk 5]
```diff
@@ -32,12 +38,16 @@ runs:
...
+        version-file: ${{ inputs.tool-versions == 'true' && '.tool-versions' || '' }}
+        version-type: ${{ inputs.tool-versions == 'true' && 'strict' || 'range' }}
```
Comment (fahchen) [inline setup-elixir/action.yml]: https://github.com/Byzanteam/jet-actions/pull/99#discussion_r1768153018

**Review by fahchen — CHANGES_REQUESTED** [review]

File: setup-elixir/action.yml (hunk 5 above)
Comment (danewalters) [reply setup-elixir/action.yml]: https://github.com/Byzanteam/jet-storage-plugin/actions/runs/10986180757

File: setup-elixir/action.yml [hunk 6]
```diff
@@ -1,26 +1,32 @@
...
 inputs:
   fetch-depth:
     description: Number of commits to fetch
     default: 1
   submodules:
     description: Whether to checkout submodules
     default: false
+  version-file:
+    description: Version file name
+    default: ".tool-versions"
```
Comment (fahchen) [inline setup-elixir/action.yml]: 不要给 defalt，给了 default 下面两个参数就没用了

**Review by fahchen — CHANGES_REQUESTED** [review]

File: setup-elixir/action.yml (hunk 6 above)
Comment (fahchen) [reply setup-elixir/action.yml]: 给一下这两个的测试结果：
1. 填 version-file 
2. 填 otp，elixir version 的

Comment (danewalters) [reply setup-elixir/action.yml]: ok

Comment (danewalters) [reply setup-elixir/action.yml]: 已经改了

Comment (danewalters) [reply setup-elixir/action.yml]: https://github.com/Byzanteam/jet-storage-plugin/actions/runs/10986180757/job/30500961824

Comment (danewalters) [reply setup-elixir/action.yml]: https://github.com/Byzanteam/jet-storage-plugin/actions/runs/10987400381/workflow

Comment (danewalters) [reply setup-elixir/action.yml]: https://github.com/Byzanteam/jet-storage-plugin/actions/runs/10987585715/workflow

**Review by fahchen — APPROVED** [review]

**Review by vanppo — APPROVED** [review]

**Review by YkeLit — APPROVED** [review]

**Pull request merged**